src/dmd/identifier.d

```diff
@@ -56,28 +63,28 @@ public:
 
     override bool equals(RootObject o) const
     {
-        return this == o || strncmp(name, o.toChars(), len + 1) == 0;
+        return this == o || name == o.toString();
     }
 
     override int compare(RootObject o) const
     {
-        return strncmp(name, o.toChars(), len + 1);
+        return strncmp(name.ptr, o.toChars(), name.length + 1);
```

Comment:
It would be nice if we could get rid of these functions that operate on C strings.

Reply:
What should we replace `strncmp` with ?

Reply:
Using the regular comparison operators?

Reply:
Uh, didn't even think it'd work.

However doesn't it include two comparison ? I can write `"a" < "b"` but I cannot get the integer value that I need to return here. So I'd need to test for comparison as well. Am I missing something ?

Reply:
I guess.

Reply:
Or it can be extracted to a utility function, I'm sure there are other places that need this as well.